domain/src/main/java/edu/nextstep/camp/calculator/domain/validator/CalculatorInputValidator.kt

```diff
@@ -0,0 +1,41 @@
+package edu.nextstep.camp.calculator.domain.validator
+
+import edu.nextstep.camp.calculator.domain.util.CalculatorInputUtil
+
+class CalculatorInputValidator {
+    companion object {
+        fun isValidOperationMark(text: String): Boolean {
+            return if (CalculatorInputUtil.isOperationMarkRegex(text)) {
+                true
+            } else {
+                throw IllegalArgumentException("The Mark is not operation mark use: [× - + ÷]")
+            }
+        }
+
+        private fun isValidOperationNumber(text: String): Boolean {
+            return if (CalculatorInputUtil.isNumberRegex(text)) {
+                true
+            } else {
+                throw IllegalArgumentException("The Number is not calculatable use: [0-9]")
+            }
+        }
+
+        fun checkIsInputEmpty(text: String) {
+            if (text.isEmpty() || text == " ") {
+                throw IllegalArgumentException("string is empty or null")
+            }
```

Comment:
```suggestion
        fun checkIsInputEmpty(text: String) {
            if (text.isEmpty() || text.isBlank()) {
                throw IllegalArgumentException("string is empty or null")
            }
```
isBlank 를 활용할 수 있어요.

Reply:
함수명에 check가 들어가있어서, 이 함수를 호출하면 true 혹은 false가 리턴될 것이라 기대가 되는데 exception이 발생하네요.
exception 대신 Boolean 값을 리턴하도록 변경해보면 어떨까요?

Reply:
exception은 주로 논리적인 에러가 발생했을 경우에 사용하곤 합니다.
하지만 이 경우에는 논리적인 에러와는 거리가 멀어보여요.
단순히 값의 유효성을 체크하고 있기 때문에, exception을 던지지 않는 방법으로 하는 것이 좋을 것 같아요. (이하 관련 내용 동일)

> use exceptions for logic errors, type-safe results for everything else. Don’t use exceptions as a work-around to sneak a result value out of a function.

ref. https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07